File: qb-engineer-server/qb-engineer.core/Models/UpdateLeadRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record UpdateLeadRequestModel(
    string? CompanyName,
    string? ContactName,
    string? Email,
    string? Phone,
    string? Source,
    LeadStatus? Status,
    string? Notes,
    DateTime? FollowUpDate,
    string? LostReason);

File: qb-engineer-ui/src/app/features/leads/models/update-lead-request.model.ts
import { LeadStatus } from './lead-status.type';

export interface UpdateLeadRequest {
  companyName?: string;
  contactName?: string;
  email?: string;
  phone?: string;
  source?: string;
  status?: LeadStatus;
  notes?: string;
  followUpDate?: string;
  lostReason?: string;
}
